frontend/src/pages/Navbar/actions.js

```diff
@@ -92,3 +104,47 @@ export function setIsRoot(isRoot) {
     isRoot
   };
 }
+
+export function showUserProfile() {
+  return {
+    type: SHOW_USER_PROFILE
+  };
+}
+
+export function hideUserProfile() {
+  return {
+    type: HIDE_USER_PROFILE
+  };
+}
+
+export function enableUserProfileEdit() {
+  return {
+    type: ENABLE_USER_PROFILE_EDIT
+  };
+}
+
+export function disableUserProfileEdit() {
+  return {
+    type: DISABLE_USER_PROFILE_EDIT
+  };
+}
+
+export function storeTempEmail(email) {
```

Comment:
does this have to be done via the whole redux cycle? or can we use a local state variable for this?